scripts/corelli/calibration/database.py

```diff
@@ -36,64 +37,165 @@ def has_valid_columns(table:TableWorkspace)->bool:
+    for iso_timestamp in iso_timestamps:
+        if not re.match('[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}', iso_timestamp):
+            raise ValueError('iso_timestamp entry: ' + str(iso_timestamp) + ' is not valid iso format YYYY-MM-DDTHH:MM:SS')
+
+    filename = database_path + '/manifest_corelli_' + date + '.csv'
+
+    if pathlib.Path(filename).is_file():
+        file = open(filename, 'a+')
+    else:
+        file = open(filename, 'w')
+        file.write('bankID, timestamp\n') # header
+
+    lines:str = ''
+    for t, bankID in enumerate(bankIDs):
+        lines += str(bankID) + ', ' + str(iso_timestamps[t]) + '\n'
+
+    file.write(lines)
+    file.close()
+
+
+def save_bank_table( data:Workspace, bankID:int, database_path:str, date:str,
+                     table_type:str = 'calibration' ) -> None:
     """
     Function that saves a bank calibrated TableWorkspace into a single HDF5 file
-    using corelli format:
-    database_path/bank0ID/type_corelli_bank0ID_YYYYMMDD.nxs
-    :param data input TableWorkspace data for calibrated pixels
+    using corelli format and using current date:
+    database_path/bank0ID/type_corelli_bank0ID_YYYYMMDD.nxs.h5
+    :param data input Workspace (TableWorkspace) data for calibrated pixels
     :param bankID bank number that is calibrated
     :param database_path location of the corelli database (absolute or relative)
            Example: database/corelli/ for
                     database/corelli/bank001/
                     database/corelli/bank002/
     :param table_type 'calibration', 'mask' or 'acceptance'
     """
-    filename:str = filename_bank_table(bankID, database_path, table_type)
+    verify_date_YYYYMMDD('save_bank_table', date)
+    filename:str = filename_bank_table(bankID, database_path, date, table_type)
     SaveNexusProcessed(data,filename)
 
 
-def load_bank_table( bankID:int, database_path:str, table_type:str = 'calibration' ) -> TableWorkspace:
+def load_bank_table( bankID:int, database_path:str, date:str, table_type:str = 'calibration' ) -> TableWorkspace:
     """
-    Function that loads a bank calibrated TableWorkspace from a single HDF5 file
+    Function that loads the latest bank calibrated TableWorkspace from a single HDF5 file
     using corelli format:
     database_path/bank0ID/type_corelli_bank0ID_YYYYMMDD.nxs
     :param bankID bank number that is calibrated
     :param database_path location of the corelli database (absolute or relative)
            Example: database/corelli/ for
                     database/corelli/bank001/
                     database/corelli/bank002/
+    :param date current day in YYYYMMDD format
     :param table_type 'calibration', 'mask' or 'acceptance'
     :return TableWorkspace with corresponding data
     """
-    filename:str = filename_bank_table(bankID, database_path, table_type)
+    verify_date_YYYYMMDD('load_bank_table', date)
+    filename:str = filename_bank_table(bankID, database_path, date, table_type)
```

Comment:
Again, I'm confused about the identity of `date`. How do you know there's a calibrated file with the same day-stamp as `date`?

Reply:
It's up to the user of this API. We allow the user to pass the date they think makes more sense if they want to load something in the past. They will have to look at the date they want to load in the database (ls). I could implement a 'latest' by default, but it would be easier if they just do the latter as these files don't change very often (once or twice a year).